xbmc/pvr/windows/GUIWindowPVRRecordings.cpp

```diff
@@ -144,13 +145,25 @@ bool CGUIWindowPVRRecordings::Update(const std::string &strDirectory, bool updat
     }
   }
 
-  if (bReturn && iOldCount > 0 && m_vecItems->GetObjectCount() == 0 && oldPath == m_vecItems->GetPath())
```

Comment:
GetObjectCount() is imo the right choice here, as it in opposite to Size() subtracts parent folder items (".." items) from the count returned.